src/libraries/Common/src/SourceGenerators/CSharpSyntaxUtilities.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Globalization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace SourceGenerators;
+
+internal static class CSharpSyntaxUtilities
+{
+    // Standard format for double and single on non-inbox frameworks.
+    public const string DoubleFormatString = "G17";
+    public const string SingleFormatString = "G9";
+
+    // Format a literal in C# format -- works around https://github.com/dotnet/roslyn/issues/58705
+    public static string FormatLiteral(object? value, TypeRef type)
+    {
+        if (value == null)
+        {
+            return $"default({type.FullyQualifiedName})";
+        }
+
+        if (type.TypeKind is TypeKind.Enum)
```

Comment:
Nit: this check is probably redundant since the default value is always provided as the underlying type.

Reply:
This was existing, but I'll double check it by adding a test and removing if not needed.

Reply:
It's also redundant since none of the enum types were covered by the switch statement and it would have fallen into default case which calls the same.